spec.html

```diff
@@ -33814,29 +33818,29 @@ <h1>String.prototype.padStart ( _maxLength_ [ , _fillString_ ] )</h1>
         <p>When the `padStart` method is called, the following steps are taken:</p>
         <emu-alg>
           1. Let _O_ be ? RequireObjectCoercible(*this* value).
-          1. Return ? StringPad(_O_, _maxLength_, _fillString_, ~start~).
+          1. Let _S_ be ? ToString(_O_).
+          1. Let _maxLen_ be ℝ(? ToLength(_maxLength_)).
+          1. If _fillString_ is *undefined*, let _filler_ be the String value consisting solely of the code unit 0x0020 (SPACE).
+          1. Else, let _filler_ be ? ToString(_fillString_).
+          1. Return StringPad(_S_, _maxLen_, _filler_, ~end~).
         </emu-alg>
 
         <emu-clause id="sec-stringpad" type="abstract operation">
           <h1>
             StringPad (
-              _O_: an ECMAScript language value,
-              _maxLength_: an ECMAScript language value,
-              _fillString_: an ECMAScript language value,
+              _S_: a String,
+              _maxLength_: a non-negative integer that is less than 2<sup>53</sup>,
+              _filler_: a String,
               _placement_: ~start~ or ~end~,
-            ): either a normal completion containing a String or an abrupt completion
+            ): a String
           </h1>
           <dl class="header">
           </dl>
           <emu-alg>
-            1. Let _S_ be ? ToString(_O_).
-            1. Let _intMaxLength_ be ℝ(? ToLength(_maxLength_)).
             1. Let _stringLength_ be the length of _S_.
-            1. If _intMaxLength_ &le; _stringLength_, return _S_.
-            1. If _fillString_ is *undefined*, let _filler_ be the String value consisting solely of the code unit 0x0020 (SPACE).
-            1. Else, let _filler_ be ? ToString(_fillString_).
+            1. If _maxLength_ &le; _stringLength_, return _S_.
```

Comment:
Oh no, I just noticed this is normative, because the length check happens _before_ coercing: `'aaaa'.padStart(0, { toString(){ throw 'no touching'; } })` does not throw, but would with this change.

That makes this refactoring a lot less appealing...

Reply:
ah, makes sense